server/app.ts

```diff
@@ -34,9 +46,49 @@ app.use(flash());
 	throw err;
 });
 
-// Auth needs to be the first route configured or else requests handled before it will always be unauthenticated
-import {authRoutes} from "./routes/auth";
-app.use("/auth", authRoutes);
+// Lets keep session data around
+app.use(session({
+	secret: config.secrets.session,
+	cookie: COOKIE_OPTIONS,
+	resave: false,
+	store: new MongoStore({
+		mongooseConnection: mongoose.connection,
+		touchAfter: 24 * 60 * 60 // Check for TTL every 24 hours at minimum
+	}),
+	saveUninitialized: true
+}));
+
+// Check to enable proxy or not
+if (!config.server.isProduction) {
+	console.warn("OAuth callback(s) running in development mode");
+}
+else {
+	app.enable("trust proxy");
+}
+
+// Check session secret & warn
+if (!config.sessionSecretSet) {
+	console.warn("No session secret set; sessions won't carry over server restarts");
+}
+
+// Auth needs to be the first route configured or else requests handled
+// Before it will always be unauthenticated
+const authService = config.server.services.auth;
+if (authService) {
+	externalAuth(authService);
+	console.log(`Not using built-in auth over ${authService.url}.`);
+} else {
+	localAuth();
+}
+
+// Create a redirect function to go with either method
+export async function redirectToLogin(req: express.Request, res: express.Response) {
```

Comment:
Please change this to `request` and `response` for consistency with the other express routes in `registration`.